pkg/solana/chainreader/event_read_binding.go

```diff
@@ -228,6 +270,56 @@ func (b *eventReadBinding) QueryKey(
 	return sequences, nil
 }
 
+func (b *eventReadBinding) normalizeParams(value any, itemType string) (any, error) {
+	offChain, err := b.codec.CreateType(itemType, true)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to create type: %w", types.ErrInvalidType, err)
+	}
+
+	// params can be a singular primitive value, a map of values, or a struct
+	// in the case that the input params are presented as a map of values, apply the values to the off-chain type
+	// with solana hooks
+	if err = codec.MapstructureDecode(value, offChain); err != nil {
+		return nil, fmt.Errorf("%w: failed to decode offChain value: %s", types.ErrInternal, err.Error())
+	}
+
+	return offChain, nil
+}
+
+func (b *eventReadBinding) extractFilterSubkeys(offChainParams any) (query.Expression, error) {
+	var expressions []query.Expression
+
+	for offChainKey, idx := range b.indexedSubKeys.lookup {
+		itemType := codec.WrapItemType(true, b.namespace, b.genericName+"."+offChainKey)
+
+		onChainValue, err := b.modifier.TransformToOnChain(offChainParams, itemType)
+		if err != nil {
+			return query.Expression{}, fmt.Errorf("%w: failed to apply on-chain transformation for key %s", types.ErrInternal, offChainKey)
+		}
+
+		// check that onChainValue is not zero value for type
+		if reflect.ValueOf(onChainValue).IsZero() {
+			continue
+		}
+
+		expression, err := logpoller.NewEventBySubKeyFilter(
+			idx,
+			[]primitives.ValueComparator{{Value: onChainValue, Operator: primitives.Eq}},
+		)
+		if err != nil {
+			return query.Expression{}, err
+		}
+
+		expressions = append(expressions, expression)
+	}
+
+	if len(expressions) == 0 {
+		return query.Expression{}, fmt.Errorf("%w: no subkey filters found during query creation", types.ErrInternal)
```

Comment:
Something doesn't sound right here. The only required field should be the EventSig (aka discriminator).

As I understand it, the EventSig is the "key" and the fields inside the event struct are the "subkeys". This comment makes it sound as if the plugin is required to narrow the search down by subkey. But one of the most common use cases of QueryKey is where you only search by Address & EventSig.

Hopefully this is just a semantic difference, and not an issue with the code?

Reply:
I do see that you have a separate `EventSig` filter here:
```
		logpoller.NewAddressFilter(pubKey),
		logpoller.NewEventSigFilter(b.eventSig[:]),
		subkeyFilters,
```
So I guess what you're calling a subkey is the same as what I think of as a subkey. But that means, you can't error out here when there are no subkey filters... right? I think that's a common case, even for CCIP

Reply:
Well this is for `GetLatestValue`, but you have a point. If no params exist, that is still a valid event call. I'll make that change.